ui.apps/src/main/content/jcr_root/apps/aemtraining/components/content/newscard/newscard.html

```diff
@@ -29,3 +26,15 @@ <h3>${newscard.article}</h3>
         </div>
     </div>
 </div>
+
+
+<script>
+    const path = '${newscard.fullPath @ context='scriptString'}';
+    function operation(tf) {
+        $.post('/services/reaction', {reaction: tf, path}, function (data) {
```

Comment:
Are you sure jQuery will work here? I can't see in this PR if you actually connected jQuery to the page with the `<script src="..."/>` tag or smth.

Please mind: if you self-tested your code in Edit or Preview mode, there's *another* jQuery that came in the "authoring" layer together with Dialogs and that stuff. 

But in "no-editing" mode, like in a true Publisher, or else with `?wcmmode=disabled`, jQuery can be missing unless deliberately connected to the page.

Reply:
I added it to my clientlib txt file but I guess I forgot to push it

Reply:
it was smth like this
 
#base=js
jquery-3.6.0.min.js

in my js.txt

Reply:
Okay but anyway.
We don't add to our own repositories code we didn't write. jQuery isn't something we wrote)
For such occasions, there's Maven to download 3rd party jar files on the fly.. and in front-end, there are similar building frameworks. One is Webpack.
All is for not adding files like jQuery into the project))

In your own case, you can add just add a single line in your markup, `<script src="jquery-3.6.0.min.js"></script>`
(Can be added even to the newscard.html, and will work, but best to add to the page's head.html.

Please be informed, also, that in modern UI jQuery is despised. There memes on the interned related to jQuery. So you will hardly see jQuery in frontend of a commercial AEM site (leave alone the "authoring" part, the" dialogs" -- in dialogs it is still used)

For your career advance, please find time to learn how to do jquery-0like things with vaniilla JS (*all* such things can be done without jquery. Absolutely all). 
See here, for instance https://stackoverflow.com/questions/50776445/vanilla-javascript-version-of-ajax

Reply:
I see, okay got it